From b0b554f12520130e35b19152733c06ad346fd92d Mon Sep 17 00:00:00 2001
From: Mohammed Naser <mnaser@vexxhost.com>
Date: Tue, 31 Jan 2023 17:26:33 -0500
Subject: [PATCH] Fixing cinder helm release storage bootstrap hooks

Change-Id: Ib457934ff539e4c7e19690a696415394634766d0
(cherry picked from commit a0bb05f41e13c7f7ebfbd2bcf81d315a98d9e6bb)
[ Reworded the shortlog ]
Signed-off-by: Luan Nunes Utimura <LuanNunes.Utimura@windriver.com>
---
 cinder/templates/job-backup-storage-init.yaml | 4 ++++
 cinder/templates/job-storage-init.yaml        | 5 +++++
 2 files changed, 9 insertions(+)

diff --git a/cinder/templates/job-backup-storage-init.yaml b/cinder/templates/job-backup-storage-init.yaml
index 8168b90f..62db734b 100644
--- a/cinder/templates/job-backup-storage-init.yaml
+++ b/cinder/templates/job-backup-storage-init.yaml
@@ -53,6 +53,10 @@ metadata:
   labels:
 {{ tuple $envAll "cinder" "backup-storage-init" | include "helm-toolkit.snippets.kubernetes_metadata_labels" | indent 4 }}
   annotations:
+{{- if .Values.helm3_hook }}
+    "helm.sh/hook": post-install,post-upgrade
+    "helm.sh/hook-delete-policy": before-hook-creation
+{{- end }}
     {{ tuple $envAll | include "helm-toolkit.snippets.release_uuid" }}
 spec:
   template:
diff --git a/cinder/templates/job-storage-init.yaml b/cinder/templates/job-storage-init.yaml
index badfe5fc..a2049d53 100755
--- a/cinder/templates/job-storage-init.yaml
+++ b/cinder/templates/job-storage-init.yaml
@@ -54,6 +54,11 @@ metadata:
   name: cinder-storage-init
   labels:
 {{ tuple $envAll "cinder" "storage-init" | include "helm-toolkit.snippets.kubernetes_metadata_labels" | indent 4 }}
+  annotations:
+{{- if .Values.helm3_hook }}
+    "helm.sh/hook": post-install,post-upgrade
+    "helm.sh/hook-delete-policy": before-hook-creation
+{{- end }}
 spec:
   template:
     metadata:
-- 
2.25.1

